### PR TITLE
Secure Development Guide

### PR DESCRIPTION
**Pull request by whitneymeredith**
Closes #[DOC-660](https://getpantheon.atlassian.net/jira/software/projects/DOC/boards/186?selectedIssue=DOC-660)

## Summary

**[Secure Development on Pantheon](https://pantheon.io/docs/guides/secure-development)** - Gathered existing docs on secure development into a guide.

## Effect

The following changes are already committed:

*
*

## Remaining Work and Prerequisites

The following changes still need to be completed:

- [ ] List any outstanding work here

### Dependencies and Timing

- [ ] Other prerequisites that must be completed before merging this PR

**Release**:
- [ ] When ready
- [ ] After date: $DATE

--------------------------------------------------

## Post Launch

**Do not remove** - To be completed by the docs team upon merge:

- [ ] Redirect `/docs/old-path/` => `/docs/new-path/` (if applicable)
- [ ] Include/exclude pages ^ respectively within docs search service provider (if applicable)
- [ ] For Heroes - add a props post to the [discussion board](https://discuss.pantheon.io/c/pantheon-platform/documentation/17).
- [ ] Remove from the [project board](https://github.com/pantheon-systems/documentation/projects/14)

### CODE REVIEW

**Comment by pantheondocs** [comment]
Lighthouse Audit Results:

| Description | [Live](https://pantheon.io/docs) | [pr-7495](https://pr-7495--pantheon-docs.my.pantheonfrontend.website/docs) | Difference |
| --- | --- | --- | --- |
| Performance Score | `53` | `51` | `-2` |
| Best Practice Score | `86` | `86` | `0` |
| SEO Score | `96` | `96` | `0` |
| Accessibility Score | `98` | `98` | `0` |

Lighthouse Audit Reports:

- [`pr-7495` HTML Report](https://circleci.com/gh/pantheon-systems/documentation/39454/artifacts/0//lighthouse_data/lighthouse-audit-dev.html)
- [`Pantheon Docs` HTML Report](https://circleci.com/gh/pantheon-systems/documentation/39454/artifacts/0//lighthouse_data/lighthouse-audit-reference.html)


[Preview Environment](https://pr-7495--pantheon-docs.my.pantheonfrontend.website/docs)

**Comment by whitneymeredith** [comment]
Redirects needed:

docs/guides/two-factor-authentication —> /guides/secure-development/two-factor-authentication

docs/ssh-tunnels —> /guides/secure-development/ssh-tunnels

docs/security —> /guides/secure-development/security-tool

docs/secure-runtime-access —> /guides/secure-development/secure-runtime-access

docs/private-paths —> /guides/secure-development/private-paths

docs/phpinfo —> /guides/secure-development/phpinfo

**Comment by whitneymeredith** [comment]
[Preview](https://pr-7495--pantheon-docs.my.pantheonfrontend.website/docs/guides/secure-development)

File: source/content/guides/secure-development/03-security-tool.md [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+![Lock environment](..//../../images/dashboard/lock-environment.png)
+
+## Password Protect Your Site's Environments
+
+You have the ability to password protect any of the available environments.
+
+1. Select the environment (for example, <span class="glyphicon glyphicon-wrench"></span> **Dev**).
+
+1. Select <span class="glyphicon glyphicon-keys"></span> **Security**.
```
Comment (wordsmither) [inline source/content/guides/secure-development/03-security-tool.md:34]: The icon isn't appearing.

File: source/content/guides/secure-development/05-two-factor-authentication.md [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+
+1. Configure the **Options** section (optional) under the SSO/SAML Settings:
+
+   - Select the **Create user if not exists** checkbox if you want users to be auto-created.
+
+   - Select the **Keep Local login** checkout if you still want to use the normal WP login form, otherwise you will always be using OneLogin to authenticate.
+
+   ![TFA OneLogin Options](../../../images/tfa-wp-onelogin-options.png)
+
+1. Configure the **Attribute Mapping** in the SSO/SAML Settings of the WordPress Admin with the values shown below. Values are case-sensitive.
```
Comment (wordsmither) [inline source/content/guides/secure-development/05-two-factor-authentication.md:172]: The values need to be included as text, for accessibility purposes.  I would have done so myself, but wasn't sure if they required any placeholder values.  Also, once the values are included as text, you may not need the image anymore.

File: source/content/guides/secure-development/05-two-factor-authentication.md [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+
+![TFA OneLogin New User](../../../images/tfa-onelogin-new-user.png)
+
+#### Drupal Instructions
+
+1. Install and enable the GitHub version of the [OneLogin SAML](https://github.com/onelogin/drupal-saml) module on your Drupal site. This module is eventually intended to live on Drupal.org as the [2.x branch of the OneLogin project](https://www.drupal.org/project/onelogin).
+
+1. Set the `$_SERVER['SERVER_PORT']` value in `settings.php` according to [these instructions](/server_name-and-server_port). This change is necessary to have SAML use the appropriate ports.
+
+1. Configure the OneLogin SAML module `admin/config/onelogin_saml` to match the screenshot below. Values are case-sensitive.
```
Comment (wordsmither) [inline source/content/guides/secure-development/05-two-factor-authentication.md:221]: Same as above: The values need to be included as text, for accessibility purposes.  I would have done so myself, but wasn't sure if they required any placeholder values.  Also, once the values are included as text, you may not need the image anymore.

**Comment by wordsmither** [comment]
lgtm

**Comment by EdwardAngert** [comment]
@whitneymeredith @wordsmither do you all have commits waiting for this one? is it okay if I throw a few copy edits in?

**Comment by whitneymeredith** [comment]
@EdwardAngert Feel free to make any edits. This is only waiting on redirects (see above). I didn't want to merge it and have most of our Security content 404.

**Pull request merged**